src/claim/Claim.js

```diff
@@ -0,0 +1,177 @@
+const _ = require('lodash');
+const sjcl = require('sjcl');
+const { UserCollectableAttribute, definitions } = require('@identity.com/uca');
+const { services } = require('../services');
+
+class Claim extends UserCollectableAttribute {
+  constructor(identifier, value, version) {
+    super(identifier, value, version);
+    this.initialize(identifier, value, version);
+  }
+
+  initialize(identifier, value, version) {
+    super.initialize(identifier, value, version);
+    if (!this.salt) {
+      const secureRandom = services.container.SecureRandom;
+      this.salt = sjcl.codec.hex.fromBits(sjcl.hash.sha256.hash(secureRandom.wordWith(64)));
+    }
+  }
+
+  initializeAttestableValue() {
+    const { value } = this;
+    const definition = this.version
+      ? _.find(definitions, { identifier: this.identifier, version: this.version })
+      : _.find(definitions, { identifier: this.identifier });
+
+    // Trying to construct UCA with a existing attestableValue
+    const parsedAttestableValue = Claim.parseAttestableValue(value);
+    if (parsedAttestableValue.length === 1) {
+      // This is a simple attestableValue
+      this.timestamp = null;
+      this.salt = parsedAttestableValue[0].salt;
+      const ucaValue = parsedAttestableValue[0].value;
+      this.value = _.includes(['null', 'undefined'], ucaValue) ? null : ucaValue;
+    } else {
+      const ucaValue = {};
```

Comment:
@brunocarneiro this method looks long (more than 25 lines), but it is readable for me. So it is up to you if you want to split it (maybe the _for block_ can become another method). This is just a suggestion.

Reply:
Because this part is legacy I prefer keeping the way it is.